aws-gitlab-main.yml

```diff
@@ -75,17 +75,60 @@ Parameters:
   GitLabDBMasterUserPasswordParameter:
     Type: String
     Description: Password for the master user. At least 8 characters. No slash, at symbol, or parentheses. 
+    NoEcho: True 
 
   GitLabAllocatedStorageParameter:
     Type: String
     Description: Amount of storage in gigabytes to be allocated for the database intsance
     Default: '100' #minimum for PostgreSQL 
 
   #ELASTICACHE/REDIS
-  CacheNodeTypeParameter:
+  GitLabCacheNodeTypeParameter:
     Type: String
     Description: Node type;  Multi-AZ not supported on T1 instances
     Default: cache.t3.medium 
+  
+  #BASTION HOSTS
+  GitLabBastionSecurityGroupCIDRParameter:
+    Type: String
+    Description: Enable SSH access from what IP address(es). For stricter security, specify a single IP 
+    Default: 0.0.0.0/0 
+
+  BastionHostAAvailabilityZoneParameter:
+    Type: AWS::EC2::AvailabilityZone::Name
+    Description: AZ for Bastion Host A
+    
+  GitLabBastionHostAInstanceTypeParameter:
+    Type: String
+    Description: EC2 instance type for Bastion Host A
+    Default: t2.micro 
+
+  GitLabBastionHostAAMIParameter:
+    Type: AWS::EC2::Image::Id
+    Description: ID of AMI for Bastion Host A
+    Default: Ubuntu-Server-18.04-LTS-HVM
+
+  GitLabBastionHostAKeyNameParameter:
+    Type: String
+    Description: Name of Key Pair for Bastion Host A
+
+  BastionHostBAvailabilityZoneParameter:
+    Type: AWS::EC2::AvailabilityZone::Name
+    Description: AZ for Bastion Host B
+
+  GitLabBastionHostBInstanceTypeParameter:
+    Type: String
+    Description: EC2 instance type for Bastion Host B
+    Default: t2.micro 
+
+  GitLabBastionHostBAMIParameter:
+    Type: AWS::EC2::Image::Id
```

Comment:
Look at the very last piece of this:
https://aws.amazon.com/blogs/compute/query-for-the-latest-amazon-linux-ami-ids-using-aws-systems-manager-parameter-store/

Implement that

Reply:
make sure you put the image id as value of ssm parameter